github-review.el

```diff
@@ -136,6 +148,32 @@ CALLBACK to call back when done."
             :callback callback
             :errorback (lambda (&rest _) (message "Error talking to GitHub"))))
 
+(defun github-review-get-commit (commit-alist needs-diff callback)
+  "Get a pull request or its diff.
+COMMIT-ALIST is an alist representing a COMMIT,
+NEEDS-DIFF t to return a diff nil to return the commit object
+CALLBACK to call back when done."
+  (ghub-get (github-review-format-commit-url 'get-commit commit-alist)
+            nil
+            :unpaginate t
+            :headers (if needs-diff github-review-diffheader '())
+            :auth 'github-review
+            :host (github-review-api-host commit-alist)
+            :callback callback
+            :errorback (lambda (&rest _) (message "Error talking to GitHub"))))
+
+
+(defun github-review-get-commit-object (commit-alist callback)
+  "Get a commit object given COMMIT-ALIST an alist recommitesenting a COMMIT.
+CALLBACK is called with the result"
+  (github-review-get-commit commit-alist nil callback))
+
+(defun github-review-get-commit-diff (commit-alist callback)
+  "Get the diff for a commit, given COMMIT-ALIST an alist recommitesenting a COMMIT.
+CALLBACK is called with the result"
+  (github-review-get-commit commit-alist t callback))
```

Comment:
I think there should be just one getter-function `github-review-get-object` that
accepts `object-id-alist` from which it and its callees find out what
kind of `PR` or `COMMIT` the object is.
E.g `github-review-get-commit` is different from `github-review-get-pr` only that
it passes in `alist` `sha`-keyed value rather than `num`-keyed one to
`github-review-format-commit-url`.
So which one is passed needs checking and then to conduct a respecive specific branch of logics.